pilot/pkg/model/mesh.go

```diff
@@ -0,0 +1,785 @@
+	// This IP is expected to be reachable from __THIS__ pilot. No distinction
+	// is being made for directly reachable service instances versus those
+	// behind a VIP. Istio's health discovery service will ensure that this
+	// endpoint's capacity is correctly reported accordingly.
+	IP destinationAttribute = "destination.ip"
+
+	// PORT represents the recipient port on the server IP address, Example: 443
+	PORT destinationAttribute = "destination.port"
+
+	// DOMAIN represents the domain portion of the service name, excluding
+	// the name and namespace, example: svc.cluster.local
+	DOMAIN destinationAttribute = "destination.domain"
+
+	// USER represents the __CANONICAL__ Mesh user running the destination
+	// application, example: service-account-foo. The user's identity is
+	// assumed to be trusted within the entire mesh, irrespective of which
+	// platform, cluster or availability zone this instance runs in.
+	USER destinationAttribute = "destination.user"
+)
+
+var (
+	destinationAttrSet = map[string]bool{
+		UID.stringValue():       true,
+		SERVICE.stringValue():   true,
+		NAME.stringValue():      true,
+		NAMESPACE.stringValue(): true,
+		IP.stringValue():        true,
+		PORT.stringValue():      true,
+		DOMAIN.stringValue():    true,
+		USER.stringValue():      true,
+	}
+
+	canonicalIstioDomain = "svc.cluster.local"
+	svcRegexp            = regexp.MustCompile(
+		"([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\\-]*[a-zA-Z0-9])\\.([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\\-]*[a-zA-Z0-9])\\.svc\\.cluster\\.local")
+	domainRegexp = regexp.MustCompile(
+		"([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\\-]*[a-zA-Z0-9])\\.([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\\-]*[a-zA-Z0-9])*")
+)
+
+type destinationAttribute string
+
+// Mesh is a platform independent abstraction used by Controllers
+// for maintaining a list of service endpoints used by this Pilot.
+//
+// Controllers under https://github.com/istio/istio/tree/master/pilot/platform
+// update the list of service endpoints in this Mesh with those
+// available in a platform specific service registry.
+//
+// Under the hoods, the Mesh implements the necessary logic required for endpoint
+// discovery based on routing rules and endpoint subsets.
+// See https://github.com/istio/api/search?q=in%3Afile+"message+Subset"+language%3Aproto
+// This logic includes comparing the updated list provided by Controllers with
+// what this view holds and accordingly updating internal structures used for
+// endpoint discovery.
+type Mesh struct {
+
+	// allEndpoints is a map of the UID to the server instance. The UID is expected
+	// to be unique across the mesh. Mesh uses this map during updates from
+	// Controllers to locate the previous representation  of the endpoint and check
+	// for changes to its metadata that could result in changes to the subsetEndpoints.
+	allEndpoints map[string]*Endpoint
+
+	// subsetDefinitions holds the metadata, basically the set of lable name values
```

Comment:
label.

You may include an example key here ( and other places ), easier to read. I assume it's the same subset as defined in the istio API - but is it qualified, etc.